internal/provider/user_resource.go

```diff
@@ -274,6 +275,23 @@ func (r *UserResource) Read(ctx context.Context, req resource.ReadRequest, resp
 	data.LoginType = types.StringValue(string(user.LoginType))
 	data.Suspended = types.BoolValue(user.Status == codersdk.UserStatusSuspended)
 
+	// Also query by username to check for deletion or username reassignment
```

Comment:
This comment should say why this is necessary from an API point of view. "The user-by-ID API always returns a user with no indication in the response that it's deleted or not, but the user-by-username API will never return deleted users."